### PR TITLE
Update lucene to version 8.11.2

### PR DESCRIPTION
**Pull request by JJK96**
This gave access to some new features in Lucene, such as Regular Expression search. This is a major refactor because I updated Lucene 5 major versions.

I tested several languages, English, Czech, Chinese, Japanese, Thai and search works in these languages. I am not capable to test if the stemming is good for all languages, so some more testing by native speakers is necessary.

### CODE REVIEW

File: src/main/java/org/apache/lucene/analysis/AbstractBookTokenFilter.java [hunk 1]
```diff
@@ -17,7 +17,7 @@
  * © CrossWire Bible Society, 2008 - 2016
  *
  */
-package org.crosswire.jsword.index.lucene.analysis;
+package org.apache.lucene.analysis;
```
Comment (tuomas2) [inline src/main/java/org/apache/lucene/analysis/AbstractBookTokenFilter.java:20]: why are these moved away from our namespace (org.crosswire)?

File: src/main/java/org/apache/lucene/analysis/ArabicLuceneAnalyzer.java [hunk 2]
```diff
@@ -26,14 +26,14 @@
...
-import org.apache.lucene.analysis.ar.ArabicLetterTokenizer;
 import org.apache.lucene.analysis.ar.ArabicNormalizationFilter;
 import org.apache.lucene.analysis.ar.ArabicStemFilter;
+import org.apache.lucene.analysis.standard.StandardTokenizer;
 import org.apache.lucene.util.Version;
 
 /**
  * An Analyzer whose {@link TokenStream} is built from a
- * {@link ArabicLetterTokenizer} filtered with {@link LowerCaseFilter},
+ * {@link StandardTokenizer} filtered with {@link LowerCaseFilter},
```
Comment (tuomas2) [inline src/main/java/org/apache/lucene/analysis/ArabicLuceneAnalyzer.java:36]: - [ ] arabic need to be tested

File: src/main/java/org/apache/lucene/analysis/AbstractBookTokenFilter.java (hunk 1 above)
Comment (JJK96) [reply src/main/java/org/apache/lucene/analysis/AbstractBookTokenFilter.java:20]: Because I needed to access some protected methods of Lucene classes in order to implement AbstractBookAnalyzer. Access to protected is only allowed in the same namespace.

Comment (tuomas2) [reply src/main/java/org/apache/lucene/analysis/AbstractBookTokenFilter.java:20]: Hmm then solution is somewhat hacky. Options to consider:
1. Fork lucene analysis lib and remove protected from that particular class (and make upstream PR). Use fork while it is needed.
2. Maybe protected is for a reason? Use some other way if lib author suggest something.
3. Accept hackyness and just leave it like this.

Comment (tuomas2) [reply src/main/java/org/apache/lucene/analysis/AbstractBookTokenFilter.java:20]: I found out that there's public abtract class StopWordAnalyzerBase that probably could be used as a base class. At least that is the baseclass within lucene core lib that is used for per-language classes there.

Comment (tuomas2) [reply src/main/java/org/apache/lucene/analysis/AbstractBookTokenFilter.java:20]: Question arises are all custom per-language analyzers still really needed or could we simplify code by using analyzers from lucene core directly. 

AbstractBookAnalyzer carries book info and it is passed to some filter classes, but any of those does not seem to use that information. I am having a feeling that all that could be simplified greatly.

Comment (JJK96) [reply src/main/java/org/apache/lucene/analysis/AbstractBookTokenFilter.java:20]: I agree, I'll look into it

**Comment by tuomas2** [comment]
@JJK96 added you write access to this repository (and AndBible dev team members too, similar to AndBible repository. Pushed what we have to update_lucene branch locally and created #16 which will replace this PR. You can push your commits directly to update_lucene branch here.

**Comment by tuomas2** [comment]
So summarizing, I would like that we try to:
- Remove AbstractBookAnalyzer alltogether, and all custom analyzers that are based on that.
- Use StopwordAnalyzer as a baseclass for our custom analyzers (KeyAnalyzer etc)
- Modify properties file / factory accordingly to use classes from core and other libs.
- Change filter classes (used by some analyzers like KeyAnalyzer) so that they do not store book (as it does not seem to be used)
